src/expr/udf/src/external.rs

```diff
@@ -139,21 +139,17 @@ impl ArrowFlightUdfClient {
     }
 
     async fn call_internal(&self, id: &str, input: RecordBatch) -> Result<RecordBatch> {
-        let mut output_stream = self.call_stream(id, stream::once(async { input })).await?;
-        // TODO: support no output
-        let head = output_stream
-            .next()
-            .await
-            .ok_or_else(Error::no_returned)??;
-        let remaining = output_stream.try_collect::<Vec<_>>().await?;
-        if remaining.is_empty() {
-            Ok(head)
-        } else {
-            Ok(arrow_select::concat::concat_batches(
-                &head.schema(),
-                std::iter::once(&head).chain(remaining.iter()),
-            )?)
+        let mut output_stream = self
+            .call_stream_internal(id, stream::once(async { input }))
```

Comment:
Why to change this? `call_stream` -> `call_stream_internal`

Reply:
If the input is empty, `call_stream` (under the hood is `do_exchange`) will return no `RecordBatch`. But we expect an empty `RecordBatch` in `call_internal`. The difference is that empty chunks contains schema info. So we introduced `call_stream_internal`, which returns an `FlightRecordBatchStream` that can be converted to an empty chunk.

Reply:
I don't understand. Isn't `call_stream` directly calls `call_stream_internal`..?

Reply:
Oh, I understood. `call_stream` make the type opaque

Reply:
I think just letting `call_stream` returns `FlightRecordBatchStream` is enough. No need to have a separate function returning `impl Stream`

Reply:
Yes. Here is the call graph:

```mermaid
graph TB
    ScalarFunctions --> Call
    Call --> Call_internal
    Call_internal --> Call_stream_internal
    TableFunctions --> Call_stream
    Call_stream --> Call_stream_internal

```

Reply:
> I think just letting `call_stream` returns `FlightRecordBatchStream` is enough. No need to have a separate function returning `impl Stream`

I just don't want to expose arrow-specific type `FlightRecordBatchStream` to the caller. 😄